src/map/vending.c

```diff
@@ -315,28 +317,16 @@ int8 vending_openvending(struct map_session_data* sd, const char* message, const
 		if( index < 0 || index >= MAX_CART // invalid position
 		||  pc_cartitem_amount(sd, index, amount) < 0 // invalid item or insufficient quantity
 		//NOTE: official server does not do any of the following checks!
-		||  !sd->status.cart[index].identify // unidentified item
-		||  sd->status.cart[index].attribute == 1 // broken item
-		||  sd->status.cart[index].expire_time // It should not be in the cart but just in case
-		||  (sd->status.cart[index].bound && !pc_can_give_bounded_items(sd)) // can't trade account bound items and has no permission
-		||  !itemdb_cantrade(&sd->status.cart[index], pc_get_group_level(sd), pc_get_group_level(sd)) ) // untradeable item
+		||  !sd->cart.u.items_cart[index].identify // unidentified item
+		||  sd->cart.u.items_cart[index].attribute == 1 // broken item
+		||  sd->cart.u.items_cart[index].expire_time // It should not be in the cart but just in case
+		||  (sd->cart.u.items_cart[index].bound && !pc_can_give_bounded_items(sd)) // can't trade account bound items and has no permission
+		||  !itemdb_cantrade(&sd->cart.u.items_cart[index], pc_get_group_level(sd), pc_get_group_level(sd)) ) // untradeable item
 			continue;
 
 		sd->vending[i].index = index;
 		sd->vending[i].amount = amount;
 		sd->vending[i].value = min(value, (unsigned int)battle_config.vending_max_value);
-
-		// Player just moved item to cart and we don't have the correct cart ID yet.
-		if (sd->status.cart[sd->vending[i].index].id == 0) {
-			struct item_data *idb = itemdb_search(sd->status.cart[index].nameid);
-			char msg[256];
-
-			sprintf(msg, msg_txt(sd, 733), idb->jname);
-			clif_displaymessage(sd->fd, msg);
-			clif_skill_fail(sd, MC_VENDING, USESKILL_FAIL_LEVEL, 0);
-			return 4;
-		}
```

Comment:
hmm, maybe u forgot that we talked about this (or maybe I overlook something on inter for cart items x mysql)
even cart/inventory/storage items are saved from map-server to char/inter-server, the `id` is never be sent back by char/inter-server to map-server.
because `id` is `AUTO_INCREMENT` by MySQL table and we cannot get the recent items were added to attach the `id`.

imho, instead make this message is set as default, (again) I prefer to tell everyone that "this is the **price** you have to 'pay' if u want to use autotrade persistance.
```c++
if (battle_config.feature_autotrade && sd->status.cart[sd->vending[i].index].id == 0) {
//...
//...
}
```